### PR TITLE
SPCTR-3296 Fix: Global - Resolved an issue with Full Site Editing (FSE) in which changes applied to page templates would not be visible when File Generation was enabled.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-3296

### Screenshots

- NA

### Types of changes

- Bug fix (non-breaking change which fixes an issue) 

### How has this been tested?

- Added to the card

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-post-assets.php [hunk 1]
```diff
@@ -226,22 +226,87 @@ public function __construct( $post_id ) {
...
+	 * @param string $post_type of current template.
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function prepare_assets_for_templates_based_post_type( $post_type ) {
+		$template_slug = $post_type;
+		if ( is_singular() && is_page() ) { // will return true if you are using a static page is a homepage ( from settings > reading ).
+			// Run only if you are on the main website URL i.e example.com.
+			$template_slug = 'page';
+		} elseif ( ! ( is_front_page() && is_home() ) && is_home() ) {   // Blog page.
```
Comment (bsf-zanev) [inline classes/class-uagb-post-assets.php:298]: Will this yield us the same result?

```suggestion
		} elseif ( is_home() && ! is_front_page() ) {   // Blog page.
```

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -7536,6 +7536,11 @@ parameters:
 			count: 1
 			path: includes/blocks/wp-search/frontend.css.php
 
+		-
+			message: "#^If condition is always true\\.$#"
+			count: 1
+			path: classes/class-uagb-post-assets.php
+
```
Comment (bsf-zanev) [inline phpstan-baseline.neon:7543]: Can we remove this baseline entry?

**Review by bsf-zanev — COMMENTED** [review]
A few small ssuggestions.

File: classes/class-uagb-post-assets.php (hunk 1 above)
Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php]: Resolved

File: phpstan-baseline.neon (hunk 2 above)
Comment (AbhijitNage123) [reply phpstan-baseline.neon]: Resolved

**Review by bsf-zanev — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

File: classes/class-uagb-helper.php [hunk 3]
```diff
@@ -157,6 +157,10 @@ public function __construct() {
 
 			self::$block_list      = UAGB_Block_Module::get_blocks_info();
 			self::$file_generation = self::allow_file_generation();
+			// Condition is only needed when we are using block based theme and Reading setting is updated. 
+			if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() && isset( $_POST['option_page'] ) && 'reading' === $_POST['option_page'] && ! empty( $_POST['submit'] ) && 'Save Changes' === $_POST['submit'] ) { //phpcs:ignore WordPress.Security.NonceVerification.Missing
+				update_option( '__uagb_asset_version', time() ); // Update the asset version when reading settings is updated.
```
Comment (adi3890) [inline classes/class-uagb-helper.php:162]: should we use https://github.com/brainstormforce/ultimate-addons-for-gutenberg/blob/096774b6f278473c60bbd7aef320e25be635416b/admin-core/ajax/common-settings.php#L680

File: classes/class-uagb-helper.php [hunk 4]
```diff
@@ -157,6 +157,10 @@ public function __construct() {
 
 			self::$block_list      = UAGB_Block_Module::get_blocks_info();
 			self::$file_generation = self::allow_file_generation();
+			// Condition is only needed when we are using block based theme and Reading setting is updated. 
+			if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() && isset( $_POST['option_page'] ) && 'reading' === $_POST['option_page'] && ! empty( $_POST['submit'] ) && 'Save Changes' === $_POST['submit'] ) { //phpcs:ignore WordPress.Security.NonceVerification.Missing
```
Comment (adi3890) [inline classes/class-uagb-helper.php:161]: verify for save changes in other language.

File: classes/class-uagb-helper.php [hunk 5]
```diff
@@ -1405,6 +1409,23 @@ public static function alignment_css( $align ) {
...
 			return $align_css;
 		}
+
+		/**
+		 * Check if the current page is related to WooCommerce.
+		 *
+		 * @since x.x.x
+		 * @return boolean True if it's a WooCommerce page, false otherwise.
+		 */
+		public static function is_woocommerce_page() {
```
Comment (adi3890) [inline classes/class-uagb-helper.php:1419]: refactor this function as get_woocommerce_template

File: classes/class-uagb-helper.php [hunk 6]
```diff
@@ -1405,6 +1409,23 @@ public static function alignment_css( $align ) {
 			}
 			return $align_css;
 		}
+
+		/**
+		 * Check if the current page is related to WooCommerce.
+		 *
+		 * @since x.x.x
+		 * @return boolean True if it's a WooCommerce page, false otherwise.
```
Comment (adi3890) [inline classes/class-uagb-helper.php:1417]: update return as per modified function

File: classes/class-uagb-post-assets.php [hunk 7]
```diff
@@ -226,22 +226,104 @@ public function __construct( $post_id ) {
...
+
+		if ( UAGB_Helper::is_woocommerce_page() ) { // Check if WooCommerce page is active.
+			$woocommerce_templates = array(
+				'is_cart'     => 'cart',
+				'is_checkout' => 'checkout',
+				'is_shop'     => 'archive-product',
+				'is_product'  => 'single-product',
+			); // WooCommerce templates to post type.
+
+			foreach ( $woocommerce_templates as $condition => $template ) {
```
Comment (adi3890) [inline classes/class-uagb-post-assets.php:281]: we can avoid foreach here using better logic in get_woocomerce_template

File: composer.json [hunk 8]
```diff
@@ -13,7 +13,8 @@
     "phpstan/phpstan": "^1.10",
     "szepeviktor/phpstan-wordpress": "^1.3",
     "php-stubs/generator": "^0.8.3",
-    "php-stubs/wordpress-stubs": "^6.2"
+    "php-stubs/wordpress-stubs": "^6.2",
+	"php-stubs/woocommerce-stubs": "^8.0"
```
Comment (adi3890) [inline composer.json:17]: check for tab spacing

File: classes/class-uagb-helper.php (hunk 3 above)
Comment (AbhijitNage123) [reply classes/class-uagb-helper.php]: Resolved

File: classes/class-uagb-helper.php (hunk 4 above)
Comment (AbhijitNage123) [reply classes/class-uagb-helper.php]: Resolved

File: composer.json (hunk 8 above)
Comment (AbhijitNage123) [reply composer.json]: Resolved

File: classes/class-uagb-helper.php (hunk 5 above)
Comment (AbhijitNage123) [reply classes/class-uagb-helper.php]: Resolved

File: classes/class-uagb-helper.php (hunk 6 above)
Comment (AbhijitNage123) [reply classes/class-uagb-helper.php]: Resolved

File: classes/class-uagb-post-assets.php (hunk 7 above)
Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php]: Resolved

**Review by adi3890 — APPROVED** [review]

**Pull request merged**